code/modules/cm_marines/dropship_equipment.dm

```diff
@@ -653,12 +655,14 @@
 	point_cost = 400
 	fire_mission_only = FALSE
 
-	update_icon()
-		if(ammo_equipped)
-			icon_state = "30mm_cannon_loaded[ammo_equipped.ammo_count?"1":"0"]"
+/obj/structure/dropship_equipment/weapon/heavygun/update_icon()
+	if(ammo_equipped)
+		icon_state = "30mm_cannon_loaded[ammo_equipped.ammo_count?"1":"0"]"
+	else
```

Comment:
https://youtu.be/cffSbDucnrM?t=7

Reply:
also this icon state management is shit. almost all of this behavior should be handled in weapon/update_icon() as they all do mostly the same, just use the initial icon_state as a base for the update_icon() to work with.